paddle/fluid/pybind/reader_py.cc

```diff
@@ -139,19 +144,44 @@ class MultiDeviceFeedReader {
   void ReadAsync() {
     for (size_t i = 0; i < readers_.size(); ++i) {
       futures_[i] = pool_->enqueue([this, i] {
-        readers_[i]->ReadNext(&ret_[i]);
-        return !ret_[i].empty();
+        try {
+          readers_[i]->ReadNext(&ret_[i]);
+          return ret_[i].empty() ? Status::kEOF : Status::kSuccess;
+        } catch (...) {
+          exceptions_[i] = std::current_exception();
+          return Status::kException;
+        }
       });
     }
   }
```

Comment:
Is it return for a void function?

Reply:
No. The return value is captured in `futures_[I]` but not the return value of `ReadAsync`.

Reply:
Sorry, I was reading incorrectly :-)